### ISSUE TITLE
npm.list callback function always returns null

### ISSUE THREAD

**Issue by imkimchi** [Manak/npm-programmatic#16]
when I try promise-then type of code, 
```
npm.list(directory)
  .then(res => console.log(res)) // it returns Unhandled rejection null
  .catch(e => console.log(e)) // it doesn't catch any error
```
and it just returns null in async-await code

```
try {
  let res = await npm.list(directory)
  console.log(res) 
} catch (e) {
  console.error(e) // null
}
```
It should be throw any error but it only returns `null`.

**Comment by Manak** [comment]
I'm going to take a look at this. Give me a while.

**Comment by wei3hua2** [comment]
#17

**Comment by huan** [comment]
I believe #17 will solve this issue, please have a look at it @Manak

**Comment by Manak** [comment]
I’m going to merge it in today

**Comment by jantuomi** [comment]
Hi, I would greatly appreciate if you could merge the fix :) Thanks for your work

**Comment by Manak** [comment]
Merged and published (v0.0.11).
